File: 6.actions/src/action_cpp_pkg/src/acc_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "rclcpp_action/rclcpp_action.hpp"

class AccServerNode : public rclcpp::Node
{
public:
    AccServerNode() : Node("acc_server_node")
    {
        RCLCPP_INFO(this->get_logger(), "Accumulate Action Server Node has been started!");
        // Create an action server
    }
};

int main(int argc, char *argv[])
{
    // Initialize the ROS 2 client library
    rclcpp::init(argc, argv);

    // Create a custom node
    auto node = std::make_shared<AccServerNode>();
    // Keep the node alive until it is shut down
    rclcpp::spin(node);
    // Destroy the node
    node.reset();

    // Shutdown the ROS 2 client library
    rclcpp::shutdown();
    return 0;
}

File: 6.actions/src/action_py_pkg/action_py_pkg/acc_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rclpy.action import ActionServer


class AccServer(Node):
    """A simple ROS2 node that creates an action server for the action "accumulate"."""

    def __init__(self):
        super().__init__("acc_server_node")
        self.get_logger().info("Accumulate Action Server Node has been started!")
        # Create an action server

    def execute_callback(self, goal_handle):
        pass


def main(args=None):
    # Initialize the ROS2 Python client library
    rclpy.init(args=args)

    # Create a ROS2 node, then spin the node to keep it alive, finally destroy the node
    node = AccServer()
    rclpy.spin(node)
    node.destroy_node()

    # Shutdown the ROS2 Python client library
    rclpy.shutdown()


if __name__ == "__main__":
    main()
